src/GraphQL.DataLoader.Tests/ApolloTracingTests.cs

```diff
@@ -0,0 +1,49 @@
+using GraphQL.Instrumentation;
+using GraphQL.Types;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.DataLoader.Tests;
+
+public class ApolloTracingTests
+{
+    // verifies that the time spent executing the data loader is included with the
+    // resolver duration, as the resolver has not 'completed' until the data loader completes
+    //
+    // note that (a) for batch data loaders, the time spent in the data loader is counted for all resolvers
+    // waiting on that data loader, and (b) any delay caused by the execution engine is counted within the
+    // resolver execution time
+    //
+    // perhaps it would be more ideal to separately record the data loader execution time within the apollo
+    // tracing output (e.g. 'extensions.tracing.execution.dataloaders'), but this does not currently happen,
+    // and the execution engine is not aware of which queued data loaders run as a single batch operation or not,
+    // making implementation difficult
+    [Fact]
+    public async Task DataLoaderTimeRecorded()
```

Comment:
This test is [unstable](https://github.com/graphql-dotnet/graphql-dotnet/actions/runs/6846713537/job/18613753677)
```bash
[xUnit.net 00:00:04.41]     GraphQL.DataLoader.Tests.ApolloTracingTests.DataLoaderTimeRecorded [FAIL]
  Failed GraphQL.DataLoader.Tests.ApolloTracingTests.DataLoaderTimeRecorded [2 s]
  Error Message:
   Shouldly.ShouldAssertException : resolverData.Duration
    should be greater than or equal to
2000000000L
    but was
1999818500L
```